src/routes/relay/entities/schema/sponsored-call.schema.spec.ts

```diff
@@ -1,59 +1,196 @@
 import { faker } from '@faker-js/faker';
+import { getMultiSendCallOnlyDeployment } from '@safe-global/safe-deployments/dist/libs';
 import * as axios from 'axios';
+import { ZodError } from 'zod';
 
+import {
+  MOCK_EXEC_TX_CALL_DATA,
+  MOCK_MULTISEND_TX_CALL_DATA,
+} from '../../../../mocks/transaction-data.mock';
 import { SponsoredCallSchema } from './sponsored-call.schema';
-
-const EXEC_TX_CALL_DATA = '0x6a761202';
+import * as txHelpers from '../../../common/transactions.helper';
 
 jest.mock('axios');
 
 describe('sponsoredCall schema', () => {
+  // chainId validation test coverage in chain-id.schema.spec.ts
+  // to address validation test coverage in address.schema.spec.ts
+
   beforeEach(() => {
     jest.clearAllMocks();
+    jest.resetAllMocks();
   });
 
-  // chainId validation test coverage in chain-id.schema.spec.ts
-  // to address validation test coverage in address.schema.spec.ts
-
-  it('should validate a valid sponsoredCall', async () => {
+  describe('should validate a valid sponsoredCall', () => {
     axios.default.get = jest.fn().mockResolvedValue({ data: 'mockSafe' });
 
-    const result = await SponsoredCallSchema.safeParseAsync({
-      chainId: '5',
-      to: faker.finance.ethereumAddress(),
-      data: EXEC_TX_CALL_DATA,
-      gasLimit: faker.random.numeric(),
+    it('should validate and extract the Safe address from a valid execTransaction call', async () => {
+      const data = {
+        chainId: '5',
+        to: faker.finance.ethereumAddress(),
+        data: MOCK_EXEC_TX_CALL_DATA,
+        gasLimit: faker.random.numeric(),
+      };
+
+      const result = await SponsoredCallSchema.safeParseAsync(data);
+
+      expect(result.success).toBe(true);
+
+      if (!result.success) {
+        return;
+      }
+
+      expect(result.data).toStrictEqual({
+        ...data,
+        safeAddress: data.to,
+      });
     });
 
-    expect(result.success).toBe(true);
+    it('should validate and extract the Safe address from a valid multisend call', async () => {
+      const chainId = '5';
+
+      const to = getMultiSendCallOnlyDeployment({
+        network: chainId,
+      })?.defaultAddress;
+
+      const data = {
+        chainId,
+        to,
+        data: MOCK_MULTISEND_TX_CALL_DATA,
+        gasLimit: faker.random.numeric(),
+      };
+
+      const result = await SponsoredCallSchema.safeParseAsync(data);
+
+      expect(result.success).toBe(true);
+
+      if (!result.success) {
+        return;
+      }
+
+      expect(result.data).toStrictEqual({
+        ...data,
+        safeAddress: '0xAecDFD3A19f777F0c03e6bf99AAfB59937d6467b',
+      });
+    });
   });
 
-  it('should not validate invalid data', async () => {
+  it('should not validate an invalid gasLimit', async () => {
     axios.default.get = jest.fn().mockResolvedValue({ data: 'mockSafe' });
 
-    for await (const data of [true, '', 'abc', '1.23', '123', 123]) {
+    for await (const gasLimit of [true, '', 'abc', '1.23', 123]) {
```

Comment:
Out of scope but just a thought for future improvements:
Now that I see this test. I find `123` as invalid `gasLimit` quite strict. 
I also don't fully grasp why it is a string / bigint. The maximum block gas limit is 30 million IIRC which easily fits into `number`
Would there be a way to accept integer `number` or string matching the regex? And we just call `toString()` on the number?

Reply:
It currently uses strings as that's what we use on the frontend. However, I can see the benefit of allowing numbers as well. Let's discuss this in the sync.